lib/internal/modules/esm/translators.js

```diff
@@ -449,65 +449,73 @@ translators.set('builtin', function builtinStrategy(url) {
-    // that resolves synchronously. To make sure we always return the identical
-    // export, we have to check again if the module already exists or not.
-    // TODO: remove CJS loader from here as well.
-    module = CJSModule._cache[modulePath];
-    if (module && module.loaded) {
-      const exports = module.exports;
-      return new ModuleWrap(url, undefined, ['default'], function() {
-        this.setExport('default', exports);
-      });
+    source = stringify(source);
+    if (shouldCheckAndPopulateCJSModuleCache) {
+      // A require call could have been called on the same file during loading and
+      // that resolves synchronously. To make sure we always return the identical
+      // export, we have to check again if the module already exists or not.
+      // TODO: remove CJS loader from here as well.
+      module = CJSModule._cache[modulePath];
+      if (module && module.loaded) {
+        const exports = module.exports;
+        return new ModuleWrap(url, undefined, ['default'], function() {
+          this.setExport('default', exports);
+        });
+      }
     }
-  }
-  try {
-    const exports = JSONParse(stripBOM(source));
-    module = {
-      exports,
-      loaded: true,
-    };
-  } catch (err) {
-    // TODO (BridgeAR): We could add a NodeCore error that wraps the JSON
-    // parse error instead of just manipulating the original error message.
-    // That would allow to add further properties and maybe additional
-    // debugging information.
-    err.message = errPath(url) + ': ' + err.message;
-    throw err;
-  }
-  if (shouldCheckAndPopulateCJSModuleCache) {
-    CJSModule._cache[modulePath] = module;
-  }
-  cjsCache.set(url, module);
-  return new ModuleWrap(url, undefined, ['default'], function() {
-    debug(`Parsing JSONModule ${url}`);
-    this.setExport('default', module.exports);
-  });
-});
+    try {
+      const exports = parser(stripBOM(source));
+      module = {
+        exports,
+        loaded: true,
+      };
+    } catch (err) {
+      // TODO (BridgeAR): We could add a NodeCore error that wraps the JSON
+      // parse error instead of just manipulating the original error message.
+      // That would allow to add further properties and maybe additional
+      // debugging information.
+      err.message = errPath(url) + ': ' + err.message;
+      throw err;
+    }
+    if (shouldCheckAndPopulateCJSModuleCache) {
+      CJSModule._cache[modulePath] = module;
+    }
+    cjsCache.set(url, module);
+    return new ModuleWrap(url, undefined, ['default'], function() {
+      debug(`Parsing JSONModule ${url}`);
+      this.setExport('default', module.exports);
+    });
+  };
+}
+
+translators.set('json', jsonStrategyBuilder(JSONParse));
+translators.set('jsonc', jsonStrategyBuilder(function jsoncParse(source) {
+  // eslint-disable-next-line node-core/no-unescaped-regexp-dot
+  return JSONParse(RegExpPrototypeSymbolReplace(/("(?:\\.|[^"\\])*?")|(\/\*[^]*?\*\/)|\/\/.*/g, source, '$1'));
```

Comment:
A comment describing what this regex is looking for would be helpful for folks coming fresh into the code.

Reply:
Also, might be a good idea to move the regex into a constant